src/util/queryBankBalances.ts

```diff
@@ -0,0 +1,21 @@
+import { QueryClient, createProtobufRpcClient } from '@cosmjs/stargate';
+import { Tendermint34Client } from '@cosmjs/tendermint-rpc';
+import { QueryClientImpl } from 'cosmjs-types/cosmos/bank/v1beta1/query';
+import type { Coin } from '@cosmjs/stargate';
+import type { HttpEndpoint } from '@cosmjs/tendermint-rpc';
+
+export const queryBankBalances = async (
+  address: string,
+  rpc: HttpEndpoint,
+): Promise<Coin[]> => {
+  const tendermint = await Tendermint34Client.connect(rpc);
+  const queryClient = new QueryClient(tendermint);
+  const rpcClient = createProtobufRpcClient(queryClient);
+  const bankQueryService = new QueryClientImpl(rpcClient);
+
+  const { balances } = await bankQueryService.AllBalances({
```

Comment:
**scaling consideration**: How often do we do this? I gather from @arirubinstein that we're supposed to make no more than 2 RPC queries per second.

I gather @michaelfig plans to provide rate limiting in `@agoric/casting` (IOU an issue #) Do we have an issue about migrating this code to use it? Cite it from a TODO?

Reply:
Yea my hope is this could unblock us for now and we can swap in the improved casting stuff later, just added a comment here.

Reply:
filed:
 - https://github.com/Agoric/agoric-sdk/issues/6870